pandas/tests/series/test_indexing.py

```diff
@@ -275,6 +275,38 @@ def test_getitem_generator(self):
         assert_series_equal(result, expected)
         assert_series_equal(result2, expected)
 
+    def test_setitem_generator(self):
+        bool_idx = self.series > 0
+        idces = self.series[bool_idx].index
+
+        values = range(bool_idx.sum())
+
+        expected = self.series.copy()
+        expected[bool_idx] = values
+
+        # list of labels:
+        s1 = self.series.copy()
+        s1[iter(idces)] = values
+        assert_series_equal(s1, expected)
+
+        # list of labels with .loc:
+        s2 = self.series.copy()
+        s2.loc[iter(idces)] = values
+        assert_series_equal(s2, expected)
+
+        import pytest
```

Comment:
imports go at the top. make 2 separate (non-class based) functions, the 2nd one which will xfail.